### PR TITLE
first crack at namespaced resolvers

### PR DESCRIPTION
**Pull request by ecwyne**

### CODE REVIEW

File: src/lib/mapResolvers.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    for (const type of Object.keys(resolvers)) {
+      for (const field of Object.keys(resolvers[type])) {
+        const fn = resolvers[type][field];
+        resolvers[type][field] = (root, args, context, info) =>
+          fn(root, args, context[namespace], info);
+      }
+    }
+    return resolvers;
+  }
+};
```
Comment (corycook) [inline src/lib/mapResolvers.js]: I have been testing something similar in my environment for a little while and something that I noticed is that this hides errors when the user provides an invalid value for a resolver.

```javascript
const resolvers = {
  TypeName: {
    getName: 'not a resolver',
  },
};
```

Apollo will throw an error during startup; however, wrapping the method hides this issue until the user attempts to execute this handler.

In my case I forgot to extract the parent resolvers:

_MyInterface.js_
```javascript
module.exports = { schema, resolvers };
```

_MyType.js_
```javascript
import MyInterface from './MyInterface';

const schema = gql`
  type MyType implements MyInterface {
    ...
  }
`;

const resolvers = Object.assign({ 
}, MyInterface); // should be MyInterface.resolvers

module.exports = { schema, resolvers };
```

Comment (ecwyne) [reply src/lib/mapResolvers.js]: @corycook would ensuring that the resolver is a `Function` be enough of a check?

```javascript
export default (namespace, resolvers) => {
  if (resolvers instanceof Object) {
   for (const type of Object.keys(resolvers)) {
     for (const field of Object.keys(resolvers[type])) {
       const fn = resolvers[type][field];
       if (typeof fn !== 'function'){
         throw new Error(`Expected Function for ${type}.${field} resolver but received ${typeof fn}`)
       }
       resolvers[type][field] = (root, args, context, info) =>
         fn(root, args, context[namespace], info);
     }
   }
   return resolvers;
  }
};
```

Comment (corycook) [reply src/lib/mapResolvers.js]: I think this check is sufficient, but you might want to just forward the value to Apollo and let them handle showing the error message in the case that the value is not a function.

Comment (corycook) [reply src/lib/mapResolvers.js]: Then again, throwing the error here might help with unit testing.

**Comment by jlengstorf** [comment]
@corycook @ecwyne Are we calling this good to go?

**Comment by ecwyne** [comment]
I believe this is good to go!

**Comment by jlengstorf** [comment]
Alright — merging.

FYI, I turned on branch protection and I'm requiring PR reviews for everyone, including administrators. So if one of you wouldn't mind giving the two housekeeping PRs I just opened a quick review, I'd appreciate it. :)

**Pull request merged**